lib/types/index.d.ts

```diff
@@ -1,98 +1,130 @@
 declare module 'node-dependency-injection' {
-    export class ContainerBuilder {
-        constructor ();
+    // Only for typings exports
 
-        addCompilerPass (compilerPass: any, ...args: any[]): void;
+    export type PassConfigHook = 'beforeOptimization'|'optimize'|'beforeRemoving'|'remove'|'afterRemoving';
 
-        compile (): void;
+    export type Parameter = string|boolean|object|any[];
 
-        findDefinition (key: any): any;
+    export type Argument = Reference|PackageReference|any;
 
-        findTaggedServiceIds (name: any): any;
+    export interface Extension {
+        load: Function;
+    }
 
-        get (id: any): any;
+    interface Logger {
+        warn(message?: any, ...optionalParams: any[]): void;
+    }
 
-        getDefinition (key: any): any;
+    export class InstanceManager {
+        constructor (containerBuilder: ContainerBuilder, definitions: Map<string, Definition>, alias: Map<string, string>);
 
-        getParameter (key: any): any;
+        getInstance (id: string, bypassPublic?: boolean): any;
 
-        has (key: any): any;
+        getInstanceFromDefinition (definition: Definition): any;
+    }
 
-        hasDefinition (key: any): any;
+    // Package exports
+    export class ContainerBuilder {
+        constructor (containerReferenceAsService?: boolean, defaultDir?: string|null);
 
-        hasParameter (key: any): any;
+        readonly defaultDir: string;
+        readonly containerReferenceAsService: boolean;
+        readonly definitions: Map<string, Definition>;
+        readonly instanceManager: InstanceManager;
+        readonly extensions: Extension[];
+        readonly services: Map<string, any>;
+        frozen: boolean;
+        logger: Logger;
 
-        isSet (id: any): any;
+        addCompilerPass (compilerPass: any, type?: PassConfigHook, priority?: number): void;
 
-        register (id: any, ...args: any[]): any;
+        compile (): void;
 
-        registerExtension (extension: any): void;
+        findDefinition (key: string): Definition;
```

Comment:
findDefinition method describes return value as Promise, but get method of Map instance isn't asynchronous. getDefinition which also uses get method from this._definitions describes correct return value - Definition. Please correct me if I'm wrong. Maybe I miss something

Reply:
you are right. no promise here